velox/vector/BaseVector.cpp

```diff
@@ -847,5 +847,51 @@ BufferPtr BaseVector::sliceBuffer(
   return ans;
 }
 
+std::string printNulls(const BufferPtr& nulls, vector_size_t maxBitsToPrint) {
+  VELOX_CHECK_GE(maxBitsToPrint, 0);
+
+  vector_size_t totalCount = nulls->size() * 8;
+  auto* rawNulls = nulls->as<uint64_t>();
+  auto nullCount = bits::countNulls(rawNulls, 0, totalCount);
+
+  std::stringstream out;
+  out << nullCount << " out of " << totalCount << " rows are null";
+
+  if (nullCount) {
+    out << ": ";
+    for (auto i = 0; i < maxBitsToPrint && i < totalCount; ++i) {
+      out << (bits::isBitNull(rawNulls, i) ? "n" : ".");
+    }
+  }
+
+  return out.str();
+}
+
+std::string printIndices(
+    const BufferPtr& indices,
+    vector_size_t maxIndicesToPrint) {
+  VELOX_CHECK_GE(maxIndicesToPrint, 0);
+
+  auto* rawIndices = indices->asMutable<vector_size_t>();
```

Comment:
Using mutable here gives the impression this will be written to. But it is read-only. Is it possible to use something else here like asRange(..) on the BufferPtr ?

Reply:
Nice catch. Changes to `as< vector_size_t >`.